docs/server_config/nginx.conf.dist

```diff
@@ -0,0 +1,175 @@
+
+    location / {
+        # Redirect pretty urls to index.php
+        try_files $uri $uri/ /index.php?$args;
+
+        # Images
+        rewrite ^/([0-9])(-[_a-zA-Z0-9-]*)?(-[0-9]+)?/.+.jpg$ /img/p/$1/$1$2$3.jpg last;
+        rewrite ^/([0-9])([0-9])(-[_a-zA-Z0-9-]*)?(-[0-9]+)?/.+.jpg$ /img/p/$1/$2/$1$2$3$4.jpg last;
+        rewrite ^/([0-9])([0-9])([0-9])(-[_a-zA-Z0-9-]*)?(-[0-9]+)?/.+.jpg$ /img/p/$1/$2/$3/$1$2$3$4$5.jpg last;
+        rewrite ^/([0-9])([0-9])([0-9])([0-9])(-[_a-zA-Z0-9-]*)?(-[0-9]+)?/.+.jpg$ /img/p/$1/$2/$3/$4/$1$2$3$4$5$6.jpg last;
+        rewrite ^/([0-9])([0-9])([0-9])([0-9])([0-9])(-[_a-zA-Z0-9-]*)?(-[0-9]+)?/.+.jpg$ /img/p/$1/$2/$3/$4/$5/$1$2$3$4$5$6$7.jpg last;
+        rewrite ^/([0-9])([0-9])([0-9])([0-9])([0-9])([0-9])(-[_a-zA-Z0-9-]*)?(-[0-9]+)?/.+.jpg$ /img/p/$1/$2/$3/$4/$5/$6/$1$2$3$4$5$6$7$8.jpg last;
+        rewrite ^/([0-9])([0-9])([0-9])([0-9])([0-9])([0-9])([0-9])(-[_a-zA-Z0-9-]*)?(-[0-9]+)?/.+.jpg$ /img/p/$1/$2/$3/$4/$5/$6/$7/$1$2$3$4$5$6$7$8$9.jpg last;
+        rewrite ^/([0-9])([0-9])([0-9])([0-9])([0-9])([0-9])([0-9])([0-9])(-[_a-zA-Z0-9-]*)?(-[0-9]+)?/.+.jpg$ /img/p/$1/$2/$3/$4/$5/$6/$7/$8/$1$2$3$4$5$6$7$8$9$10.jpg last;
+        rewrite ^/c/([0-9]+)(-[.*_a-zA-Z0-9-]*)(-[0-9]+)?/.+.jpg$ /img/c/$1$2$3.jpg last;
+        rewrite ^/c/([a-zA-Z_-]+)(-[0-9]+)?/.+.jpg$ /img/c/$1$2.jpg last;
+
+        # AlphaImageLoader for IE and fancybox
+        rewrite ^images_ie/?([^/]+)\.(jpe?g|png|gif)$ js/jquery/plugins/fancybox/images/$1.$2 last;
+
+        # Web service API
+        rewrite ^/api/?(.*)$ /webservice/dispatcher.php?url=$1 last;
+
+        # Installation sandbox
+        rewrite ^(/install(?:-dev)?/sandbox)/(.*) /$1/test.php last;
+    }
+
+    # File security
+    # .htaccess .DS_Store .htpasswd etc
+    location ~ /\. {
+        deny all;
+    }
+    # Source code directories
+    location ~ ^/(app|bin|cache|classes|config|controllers|docs|localization|override|src|tests|tools|translations|travis-scripts|vendor)/ {
+        deny all;
+    }
+    # Prevent exposing other sensitive files
+    location ~ \.(yml|log|tpl|twig|sass)$ {
+        deny all;
+    }
+
+    # Prevent injection of php files
+    location /upload {
+        location ~ \.php$ {
+            deny all;
+        }
+    }
+
+    # PHP FPM part
+    location ~ [^/]\.php(/|$) {
+        # Verify that the file exists, redirect to index if not
+        try_files $fastcgi_script_name /index.php$uri&$args;
+
+        fastcgi_index  index.php;
+
+        # Envirnoment variables for PHP
+        fastcgi_split_path_info ^(.+\.php)(/.+)$;
+
+        include       fastcgi_params;
+
+        fastcgi_param PATH_INFO       $fastcgi_path_info;
+        fastcgi_param PATH_TRANSLATED $document_root$fastcgi_path_info;
+        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+
+        # [REQUIRED EDIT] Connection to PHP-FPM - choose one
+        # fastcgi_pass 127.0.0.1:9000;
+        fastcgi_pass unix:/run/php/php7.0-fpm.sock;
```

Comment:
This _is_ a virtual host configuration. Vhost are declared with `server {}` statements on Ningx.

I suppose this value could be put as a variable on the global server scope and then be imported here. But in any case, it's up to the sysadmin who puts it into production, as it would have to be done outside of this file.

This file contains the "vhost" part only, and is to be imported by `nginx.conf` -- you'd have to declare your variable there, not here.